### PR TITLE
add weekly newsletters page

### LINKED ISSUE #245: Weekly newsletters
Page displaying weekly newsletters

### PR DESCRIPTION
**Pull request by MikaelSiidorow**
Closes #245

### CODE REVIEW

File: apps/web/src/lib/api/fetcher.ts [hunk 1]
```diff
@@ -71,6 +71,7 @@ export function getAll<
           depth: 10, // TODO: remove this when we have a better way to handle depth for example with GraphQL
           // Needs to be bigger than 1 to get media / images
           limit: 100,
+          sort: "-createdAt",
```
Comment (kahlstrm) [inline apps/web/src/lib/api/fetcher.ts:74]: Let's have this as a parameter that is passed/sort in next.js side, I don't think sorting all queries is good.

File: apps/web/src/app/[locale]/[...path]/page.tsx [hunk 2]
```diff
@@ -96,6 +99,14 @@ async function Page({ params: { path } }: Props) {
...
     }
 
+    if (page.specialPageType === "weekly-newsletter") {
+      return <WeeklyNewsletterPage />;
+    }
+
+    if (page.specialPageType === "weekly-newsletters-list") {
+      return <WeeklyNewslettersListPage />;
+    }
+
```
Comment (kahlstrm) [inline apps/web/src/app/[locale]/[...path]/page.tsx:109]: Is there a reason we have nested typing, i.e. `page.type` is now special and that special page has more than one type itself. Why not just have these in `page.type` instead? Or at least it seems like this feels like unnecesarily nested logic, and with the other pattern we can give each special page their own custom fields.

**Review by kahlstrm — CHANGES_REQUESTED** [review]

File: apps/web/src/lib/api/fetcher.ts (hunk 1 above)
Comment (MikaelSiidorow) [reply apps/web/src/lib/api/fetcher.ts:74]: Makes sense, but needed I thought it needed much refactoring for the fetcher implementation that I wanted to skip it :sweat_smile: I'll do it

File: apps/web/src/app/[locale]/[...path]/page.tsx (hunk 2 above)
Comment (MikaelSiidorow) [reply apps/web/src/app/[locale]/[...path]/page.tsx:109]: I just thought that there might be shared fields between all custom pages, like if we had route params there. I'll think about it

File: apps/web/src/lib/api/fetcher.ts (hunk 1 above)
Comment (MikaelSiidorow) [reply apps/web/src/lib/api/fetcher.ts]: Fixed

File: apps/web/src/app/[locale]/[...path]/page.tsx (hunk 2 above)
Comment (MikaelSiidorow) [reply apps/web/src/app/[locale]/[...path]/page.tsx]: I guess it makes sense to just be under `page.type`, couldn't come up with any reason we would have shared fields between all special-pages and we can do it then

**Pull request merged**